plugins/modules/source_control/github/github_repo.py

```diff
@@ -0,0 +1,246 @@
+        target = gh.get_organization(organization)
+    else:
+        target = gh.get_user()
+
+    repo = None
+    try:
+        repo = target.get_repo(name=name)
+        result['repo'] = repo.raw_data
+    except UnknownObjectException:
+        if not check_mode:
+            repo = target.create_repo(
+                name=name, private=private, description=description)
+            result['repo'] = repo.raw_data
+
+        result['changed'] = True
+
+    changes = {}
+    if check_mode or repo.raw_data['private'] != private:
+        changes['private'] = private
+    if check_mode or repo.raw_data['description'] != description:
+        changes['description'] = description
+
+    if changes:
+        if not check_mode:
+            repo.edit(**changes)
+
+        result['repo'].update({
+            'private': repo._private.value if not check_mode else private,
+            'description': repo._description.value if not check_mode else description
+        })
+        result['changed'] = True
+
+    return result
+
+
+def delete_repo(gh, name, organization=None, check_mode=False):
+    result = dict(changed=False)
+    if organization:
+        target = gh.get_organization(organization)
+    else:
+        target = gh.get_user()
+    try:
+        repo = target.get_repo(name=name)
+        if not check_mode:
+            repo.delete()
+        result['changed'] = True
+    except UnknownObjectException:
+        pass
+
+    return result
+
+
+def run_module(params, check_mode=False):
+    gh = authenticate(
+        username=params['username'], password=params['password'], access_token=params['access_token'])
+    if params['state'] == "absent":
+        args = {
+            "gh": gh,
+            "name": params['name'],
+            "organization": params['organization'],
+            "check_mode": check_mode
+        }
+        return delete_repo(**args)
+    else:
+        args = {
+            "gh": gh,
+            "name": params['name'],
+            "organization": params['organization'],
+            "private": params['private'],
+            "description": params['description'],
+            "check_mode": check_mode
+        }
+        return create_repo(**args)
+
+
+def main():
+    module_args = dict(
+        username=dict(type='str', required=False, default=None),
+        password=dict(type='str', required=False, default=None, no_log=True),
+        access_token=dict(type='str', required=False,
+                          default=None, no_log=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', required=False, default="present",
+                   choices=["present", "absent"]),
+        organization=dict(type='str', required=False, default=None),
+        private=dict(type='bool', required=False, default=False),
+        description=dict(type='str', required=False, default=''),
+    )
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
```

Comment:
```suggestion
        supports_check_mode=True,
        required_together=[('username', 'password')],
        required_one_of=[('username', 'access_token')],
        mutually_exclusive=[('username', 'access_token')],
```
This way Ansible will make sure that the correct parameters are passed.